src/main/java/org/mule/extension/db/internal/domain/connection/JdbcConnectionFactory.java

```diff
@@ -26,9 +27,18 @@
  */
 public class JdbcConnectionFactory {
 
+  /**
+   * Ensures DriverManager classloading takes place before any connection creation. It prevents a JDK deadlock that only occurs
+   * when two JDBC Connections of different DB vendors are created concurrently and the {@link DriverManager} hasn't been loaded
+   * yet. For more information, see MULE-14605.
+   */
+  static {
+    DriverManager.getLoginTimeout();
+  }
+
```

Comment:
Perhaps this could be done by Mule in order to prevent this instead of the connector's code: org.mule.runtime.module.artifact.internal.classloader.DefaultResourceInitializer